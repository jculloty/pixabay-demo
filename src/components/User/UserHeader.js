import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class UserHeader extends PureComponent {
  static propTypes = {
    user: PropTypes.object.isRequired,
  }

  render() {
    const { user } = this.props;
    return (
      <div>
        <h2>{user.name} / {user.total}</h2>
        <img src={user.url} alt="" className="user-image" />
      </div>
    );
  }
}

export default UserHeader;
