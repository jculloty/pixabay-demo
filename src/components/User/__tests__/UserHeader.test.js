import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UserHeader from '../UserHeader';

const buildComponent = function(user = { name: 'John-Doe', total: 100, url: 'www.jd.com' }) {
  return (
    <MemoryRouter>
      <UserHeader user={user} />
    </MemoryRouter>
  );
};

describe('User: UserHeader', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
