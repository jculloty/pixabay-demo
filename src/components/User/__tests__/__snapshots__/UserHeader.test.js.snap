// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User: UserHeader renders correctly 1`] = `
<div>
  <h2>
    John-Doe
     / 
    100
  </h2>
  <img
    alt=""
    className="user-image"
    src="www.jd.com"
  />
</div>
`;
